src/evm/protocol/uniswap_v4/hooks/generic_vm_hook_handler.rs

```diff
@@ -63,10 +63,15 @@ where
 {
     pub fn new(
         address: Address,
-        bytecode: Bytecode,
         engine: SimulationEngine<D>,
         pool_manager: Address,
+        // TODO are all of these necessary?
+        _all_tokens: HashMap<Bytes, Token>,
+        _account_balances: HashMap<Bytes, HashMap<Bytes, Bytes>>,
```

Comment:
I would say that account_balances is enough. it should include already this component's balance _I think_